src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -557,24 +559,32 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t,
                     uct_rc_mlx5_iface_common_config_t *mlx5_config,
                     uct_ib_iface_init_attr_t *init_attr)
 {
+    int dc = (init_attr->qp_type == UCT_IB_QPT_DCI);
     uct_ib_device_t *dev;
     ucs_status_t status;
 
+    ucs_assert((init_attr->qp_type == UCT_IB_QPT_DCI) ||
+               (init_attr->qp_type == IBV_QPT_RC));
+
     status = uct_rc_mlx5_iface_preinit(self, md, rc_config, mlx5_config,
                                        params, init_attr);
     if (status != UCS_OK) {
         return status;
     }
 
     self->rx.srq.type                = UCT_IB_MLX5_OBJ_TYPE_LAST;
-    self->rx.srq.topo                = uct_rc_mlx5_iface_srq_topo(self, md, mlx5_config);
     self->tm.cmd_wq.super.super.type = UCT_IB_MLX5_OBJ_TYPE_LAST;
     init_attr->rx_hdr_len            = UCT_RC_MLX5_MP_ENABLED(self) ?
                                        0 : sizeof(uct_rc_mlx5_hdr_t);
 
     UCS_CLASS_CALL_SUPER_INIT(uct_rc_iface_t, ops, md, worker, params,
                               rc_config, init_attr);
 
+    self->rx.srq.topo = uct_rc_mlx5_iface_srq_topo(self, mlx5_config, dc);
```

Comment:
maybe pass qp_type to uct_rc_mlx5_iface_srq_topo ?

Reply:
done